### PR TITLE
fix(loop): greenlet + async code health code

### PR DESCRIPTION
**Pull request by pavelfeldman**
- bubbles the run_until_complete to main.py
- creates loop if missing
- removes dispatcher greenlet singleton

### CODE REVIEW

File: playwright/transport.py [hunk 1]
```diff
@@ -15,42 +15,47 @@
...
         self.on_message = lambda _: None
         self._stopped = False
-
-    def run_sync(self) -> None:
-        self.loop.run_until_complete(self._run())
-
-    def run_async(self) -> None:
-        self.loop.create_task(self._run())
+        self._driver_executable = driver_executable
+        self.loop: asyncio.AbstractEventLoop
```
Comment (dgozman) [inline playwright/transport.py:29]: Probably `self._loop` with an underscore?

**Review by dgozman — APPROVED** [review]

**Comment by Mattwmaster58** [comment]
Something I just encountered is that with Python 3.7 + threads + unix, will fail with this error:
```
RuntimeError: Cannot add child handler, the child watcher does not have a loop attached
```
The solution is to make sure there's a asyncio event loop in the main thread and that the child watcher is activated:
```py
# (at file level)
asyncio.get_current_loop()
asyncio.get_child_watcher
```
(not necessary in 3.8+ as the default child watcher implementation is better)
Out of the scope this PR, but I feel there should at least be some sort of mention of this in the docs.

**Comment by Mattwmaster58** [comment]
Something I just encountered is that with Python 3.7 + threads + unix, will fail with this error:
```
RuntimeError: Cannot add child handler, the child watcher does not have a loop attached
```
The solution is to make sure there's a asyncio event loop in the main thread and that the child watcher is activated:
```py
# (at file level)
asyncio.get_current_loop()
asyncio.get_child_watcher
```
(not necessary in 3.8+ as the default child watcher implementation doesn't have this limitation)
Out of the scope this PR, but it seems like there should be maybe a mention of this somewhere? Or is ti too niche a use-case?

**Comment by pavelfeldman** [comment]
@Mattwmaster58 I'll add some tests for the Thread. Is that just running sync python on a thread use case?

**Comment by Mattwmaster58** [comment]
Yeah, an example for this setup would gunicorn + flask with endpoints that use the sync interface.

**Pull request merged**